### PR TITLE
Convert to safe-loggable logs and preconditions

### PR DESCRIPTION
**Pull request by dansanduleac**
## Before this PR

A bunch of exceptions and logs did not use safe-loggable constructs, and so would be redacted.

## After this PR
==COMMIT_MSG==
Use safe-loggable log arguments and preconditions everywhere.
==COMMIT_MSG==

### CODE REVIEW

File: auth-tokens/src/main/java/com/palantir/tokens/auth/UnverifiedJsonWebToken.java [hunk 1]
```diff
@@ -110,13 +108,13 @@ private static int countCharacter(String input, char toCount) {
...
      */
     public static UnverifiedJsonWebToken of(BearerToken token) {
-        List<String> segments = Splitter.on('.').splitToList(token.getToken());
+        String[] segments = token.getToken().split("\\.");
         Preconditions.checkArgument(
-                segments.size() == 3,
+                segments.length == 3,
                 "Invalid JWT: expected 3 segments",
-                SafeArg.of("segmentsCount", segments.size()));
+                SafeArg.of("segmentsCount", segments.length));
```
Comment (carterkozak) [inline auth-tokens/src/main/java/com/palantir/tokens/auth/UnverifiedJsonWebToken.java]: this is a hot path, worth inverting this into `if (segments !=3) { /*work + throw */}`?

**Review by carterkozak — APPROVED** [review]

File: auth-tokens/src/main/java/com/palantir/tokens/auth/UnverifiedJsonWebToken.java (hunk 1 above)
Comment (dansanduleac) [reply auth-tokens/src/main/java/com/palantir/tokens/auth/UnverifiedJsonWebToken.java]: won't the JVM figure this out eventually?

Comment (carterkozak) [reply auth-tokens/src/main/java/com/palantir/tokens/auth/UnverifiedJsonWebToken.java]: Not likely, especially if we ever encounter a bad input.

Comment (dansanduleac) [reply auth-tokens/src/main/java/com/palantir/tokens/auth/UnverifiedJsonWebToken.java]: alright, changed

**Pull request merged**